spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -532,6 +537,24 @@ public final void initialize(String name, CaseInsensitiveStringMap options) {
             Splitter.on('.').splitToList(options.get("default-namespace")).toArray(new String[0]);
       }
     }
+
+    EnvironmentContext.put(EnvironmentContext.ENGINE_NAME, "spark");
+    EnvironmentContext.put(
+        EnvironmentContext.ENGINE_VERSION, sparkSession.sparkContext().version());
+    EnvironmentContext.put(CatalogProperties.APP_ID, sparkSession.sparkContext().applicationId());
+    sparkSession
+        .sparkContext()
+        .addSparkListener(
+            new SparkListener() {
+              @Override
+              public void onOtherEvent(SparkListenerEvent event) {
+                if (event instanceof SparkListenerSQLExecutionStart) {
+                  SparkListenerSQLExecutionStart start = (SparkListenerSQLExecutionStart) event;
+                  EnvironmentContext.putLocal(
+                      SQLExecution.EXECUTION_ID_KEY(), () -> String.valueOf(start.executionId()));
```

Comment:
Doesn't this implementation defeat the purpose of using a supplier to get the value? This captures the source of the value in the supplier, which is why it needs to be threadlocal.